Improve README and add contributing guidelines (#2612)
iotaledger/trinity-wallet: README.md

    <a href="https://iota.stackexchange.com/" style="text-decoration:none;"><img src="https://img.shields.io/badge/StackExchange-9cf.svg?logo=stackexchange" alt="StackExchange"></a>
    <a href="https://raw.githubusercontent.com/iotaledger/trinity-wallet/develop/LICENSE" style="text-decoration:none;"><img src="https://img.shields.io/badge/License-Apache%202.0-green.svg" alt="Apache 2.0 license"></a>
    <a href="https://dependabot.com" style="text-decoration:none;"><img src="https://api.dependabot.com/badges/status?host=github&repo=iotaledger/trinity-wallet" alt=""></a>
</p>
      
<p align="center">
  <a href="#about">About</a> ◈
  <a href="#prerequisites">Prerequisites</a> ◈
  <a href="#installation">Installation</a> ◈
  <a href="#getting-started">Getting started</a> ◈
  <a href="#supporting-the-project">Supporting the project</a> ◈
  <a href="#joining-the-discussion">Joining the discussion</a> 
</p>

---

## About

This is the **official** IOTA wallet, which allows you to do the following:
* Create password-protected accounts to store and access your seeds
* Send transactions
* Generate addresses
* Read your balance and transaction history

> This is beta software, so there may be performance and stability issues.

Reviewer: ```suggestion
```

Trinity is no longer in beta, so this warning doesn't really apply

Please report any issues in our [issue tracker](https://github.com/iotaledger/trinity/issues/new).

## Prerequisites

To compile Trinity, you must have the following:

> * Node.js 8 or higher. We recommend the [latest LTS](https://nodejs.org/en/download/)

Reviewer: ```suggestion
* Node.js 10 or 11
```
Node 8 is probably fine to use but it has reached EOL. Some native components on desktop don't support Node 12 yet (we're awaiting fixes upstream) so I don't think we should say "or higher"

Author: Sounds good 👌 

I'll add a link to a downloads page

* [Yarn](https://yarnpkg.com/) package manager.

## Installation

To install the dependencies needed to compile both Trinity Mobile and Trinity Desktop, do the following:

1. Clone this repository

    ```
    git clone https://github.com/iotaledger/trinity-wallet.git
    ```

2. Change into the `trinity-wallet` directory

    ```
    cd trinity-wallet
    ```

3. Install the shared dependencies

    ```
    yarn && yarn deps:shared
    ```

## Getting started

> To compile Trinity, follow the instructions for either [Trinity Mobile](https://github.com/iotaledger/trinity-wallet/blob/develop/src/mobile/README.md) or [Trinity Desktop](https://github.com/iotaledger/trinity-wallet/blob/develop/src/desktop/README.md).

Reviewer: I think it might be better to have this in the "Installation" section, as the shared dependencies should be installed before installing the dependencies specific to mobile or desktop

Author: Added this to installation 👍

## Supporting the project

> If Trinity has been useful to you and you feel like contributing, consider posting a [bug report](https://github.com/iotaledger/trinity-wallet/issues/new), [feature request](https://github.com/iotaledger/trinity-wallet/issues/new) or a [pull request](https://github.com/iotaledger/trinity-wallet/pulls/).

Reviewer: ```suggestion
If Trinity has been useful to you and you feel like contributing, consider posting a [bug report](https://github.com/iotaledger/trinity-wallet/issues/new?labels=T+-+Bug&template=bug_report.md&title=), [feature request](https://github.com/iotaledger/trinity-wallet/issues/new?labels=&template=feature_request.md&title=) or a [pull request](https://github.com/iotaledger/trinity-wallet/pulls/).
```
Updated links so that they direct readers to the appropriate templates

See the [contributing guidelines](CONTRIBUTING.md) for more information.

## Joining the discussion

> If you want to get involved in the community, need help with getting setup, have any issues related with the library or just want to discuss IOTA, Distributed Registry Technology (DRT) and IoT with other people, feel free to join our [Discord](https://discord.iota.org/).

Reviewer: ```suggestion
If you want to get involved in the community, need help with getting setup, have any issues related to Trinity or just want to discuss IOTA, Distributed Ledger Technology (DLT) and IoT with other people, feel free to join our [Discord](https://discord.iota.org/).
```
